src/posix/platform/openthread-posix-config.h

```diff
@@ -203,7 +203,8 @@
  *
  */
 #ifdef __linux__
-#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE & OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
+#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE && OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE && \
+    !OPENTHREAD_POSIX_CONFIG_DAEMON_ENABLE
```

Comment:
Shouldn't we use dedicated cmake options (e.g. `OT_DAEMON`) for the ingress filter feature?